agent/engine/docker_image_manager.go

```diff
@@ -175,6 +174,10 @@ func (imageManager *dockerImageManager) addContainerReferenceToExistingImageStat
 	imageState, ok := imageManager.getImageState(container.ImageID)
 	if ok {
 		imageState.UpdateImageState(container)
+		err := imageManager.dataClient.SaveImageState(imageState)
```

Comment:
can you add some test cases for changes in this file using a test data client? you can refer to what i have in https://github.com/aws/amazon-ecs-agent/blob/boltdb/agent/engine/task_manager_data_test.go and https://github.com/aws/amazon-ecs-agent/blob/boltdb/agent/engine/data_test.go

Reply:
Done, added test cases in engine/docker_image_manager_data_test.go